Dockerfile-slim.template

```diff
@@ -31,6 +31,8 @@ RUN ARCH= && dpkgArch="$(dpkg --print-architecture)" \
     && grep " node-v$NODE_VERSION-linux-$ARCH.tar.xz\$" SHASUMS256.txt | sha256sum -c - \
     && tar -xJf "node-v$NODE_VERSION-linux-$ARCH.tar.xz" -C /usr/local --strip-components=1 --no-same-owner \
     && rm "node-v$NODE_VERSION-linux-$ARCH.tar.xz" SHASUMS256.txt.asc SHASUMS256.txt \
+    # Remove unused OpenSSL headers to save ~34MB. See this NodeJS issue: https://github.com/nodejs/node/issues/46451
+    && find /usr/local/include/node/openssl/archs -mindepth 1 -maxdepth 1 ! -name linux-$(uname -m) -exec rm -rf {} \; \
```

Comment:
I don't have strong opinions on this PR one way or another, but `uname -m` is _kernel_ architecture detection and needs to be _userspace_ architecture detection instead (either using `dpkg-architecture` or `dpkg --print-architecture`, for example).

Reply:
Not sure I understand, would you mind elaborating?
`uname -m` prints `x86_64`, which matches the headers dir called `linux-x86_64`, while `dpkg --print-architecture` prints `amd64`, which is obviously the same arch but formatted otherwise and doesn't match.

Could there be a meaningful difference between the two? That is, could `dpkg --print-architecture` print, say, `x86` while `uname -m` prints `x86_64`? Or even arm vs intel?

Reply:
For all the arm32vX builds on the downstream Docker Official Images infrastructure, `uname -m` will print `armv8l` which isn't correct for any of them (where `dpkg --print-architecture` will be `armhf` instead which likely needs to be translated to `armv7l` or whatever the appropriate header folder name is), so I'd suggest a rough `case` statement that does the translation just like the one seen further up for determining which Node.js architecture tarball to download.

Reply:
Thanks explaining @tianon!
Updated diff to use the same switch case :)

Reply:
@tianon anything else I can do to promote this?